packages/beacon-node/src/chain/validation/attestation.ts

```diff
@@ -470,7 +471,9 @@ async function validateAttestationNoSignatureCheck(
   let attDataRootHex: RootHex;
   const signature = attestationOrCache.attestation
     ? attestationOrCache.attestation.signature
-    : getSignatureFromAttestationSerialized(attestationOrCache.serializedData);
+    : !isForkPostElectra(fork)
```

Comment:
Personally would like to see the fork switching logic to be hidden in `sszBytes.ts`

Maybe a function named `getSignatureFromBeaconAttestationSerialized` in a similar fashion  as `getBlockRootFromBeaconAttestationSerialized`.

Reply:
was debating that as well but figured we only use those in network processor and this pattern was used before already so I went with explicit fork logic in validation itself
https://github.com/ChainSafe/lodestar/blob/957684fe0393b3f50087fc618fb319156ab683fd/packages/beacon-node/src/chain/validation/attestation.ts#L265-L267

Either one seems fine to me